qiskit/assembler/assemble_schedules.py

```diff
@@ -242,3 +260,52 @@ def _bundle_channel_indices(instructions):
         mem_slots.extend(mem_slot.index for mem_slot in inst.mem_slots)
         reg_slots.extend(reg.index for reg in inst.reg_slots)
     return qubits, mem_slots, reg_slots
+
+
+def _assemble_config(lo_converter: LoConfigConverter,
+                     experiment_config: Dict[str, Any],
+                     run_config: RunConfig) -> PulseQobjConfig:
+    """Assembles the QobjConfiguration from experimental config and runtime config.
+
+    Args:
+        lo_converter: The configured frequency converter and validator
+        experiment_config: schedules to assemble
+        run_config: configuration of the runtime environment
+
+    Returns:
+        The assembled PulseQobjConfig
+    """
+    qobj_config = run_config.to_dict()
+    qobj_config.update(experiment_config)
+
+    # Run config not needed in qobj config
+    qobj_config.pop('meas_map', None)
+    qobj_config.pop('qubit_lo_range', None)
+    qobj_config.pop('meas_lo_range', None)
+
+    # convert enums to serialized values
+    meas_return = qobj_config.get('meas_return', 'avg')
+    if isinstance(meas_return, MeasReturnType):
+        qobj_config['meas_return'] = meas_return.value
+
+    meas_level = qobj_config.get('meas_level', 2)
+    if isinstance(meas_level, MeasLevel):
+        qobj_config['meas_level'] = meas_level.value
+
+    # convert lo frequencies to Hz
+    qobj_config['qubit_lo_freq'] = [freq / 1e9 for freq in qobj_config['qubit_lo_freq']]
+    qobj_config['meas_lo_freq'] = [freq / 1e9 for freq in qobj_config['meas_lo_freq']]
+
+    # frequency sweep config
+    schedule_los = qobj_config.pop('schedule_los', [])
+    if len(schedule_los) == 1:
+        lo_dict = schedule_los[0]
+        q_los = lo_converter.get_qubit_los(lo_dict)
```

Comment:
Would it make sense for `q_los` and `m_los` to be in `experiment_config` (so that `_assemble_config` doesn't need the `lo_converter`)?

Reply:
Possibly? I might prefer that we didn't need `experiment_config` at all, since `_assemble_instructions` shouldn't be responsible for building any config, only `_assemble_config` should be doing that... Earlier, I had avoided doing any formatting in `_assemble_instructions`, and I returned a 3-tuple instead. Both approaches have some downsides.

I agree the `lo_converter` stuff can be improved. If you look at it, it doesn't actually do anything besides check that the frequencies are within the range provided by the backend. It's doing some very minimal validation without ever really making it clear that that's all its doing (and using multiple classes to basically `assert freq_min[i] < freq[i] < freq_max[i]`. I've been thinking for a long time that `LoConfig` should be removed entirely.

Do you think this could be a follow up discussion issue?

Reply:
:+1: good idea.